server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -758,6 +761,7 @@ public <Request extends ActionRequest, Response extends ActionResponse> void reg
         actions.register(MultiGetAction.INSTANCE, TransportMultiGetAction.class);
         actions.register(TransportShardMultiGetAction.TYPE, TransportShardMultiGetAction.class);
         actions.register(BulkAction.INSTANCE, TransportBulkAction.class);
+        actions.register(SimulateBulkAction.INSTANCE, TransportSimulateBulkAction.class);
```

Comment:
I think it makes it much easier to find stuff when they're all named the same, so could we either use `RestSimulateBulkAction` or change these to `*SimulateIngestAction`?

Reply:
This one is a little tricky, and there's actually a reason behind the difference. I named the rest action RestSimulateIngestAction because it's analogous to RestSimulatePipelineAction. But we transform the inputs to use as much of the BulkAction logic as possible. So the SimulateBulkAction/TransportSimulateBulkAction are named that way because they just extend BulkAction/TransportBulkAction. So there's going to be a disconnect whichever way we go. If I change RestSimulateIngestAction to RestSimulateBulkAction for example, it's no longer going to be easily findable by someone who is looking at the RestSimulatePipelineAction and wants to see what simulate ingest looks like.